### ISSUE TITLE
[Feature Request] Pause multiple applications

### ISSUE THREAD

**Issue by BlueManCZ** [Merrit/nyrna#4]
It would be great to allow the user to pause multiple applications at the same time. Currently, when I pause an application and then try to pause another one, the first one always gets unpaused.

**Comment by Merrit** [comment]
Thank you for the suggestion! :)

I have some ideas already of how I might implement this.

**Comment by BlueManCZ** [comment]
I just performed some testing and it seems I'm able to get PID of the focused suspended window. I don't know if this is environment-specific or if Mac / Win allow it as well. Needs testing.

1. Suspend window with Nyrna.
2. Open a terminal and execute: `sleep 5; xdotool getactivewindow getwindowpid`
3. In following 5 seconds switch to the suspended window.
4. Switch back to the terminal and check PID.

It came to my mind that something like this should be possible to code (its pseudocode):
```
pauseKeyInvoked()
  pid := getPidOfActiveWindow()
  if pid != nil
    if pid in listOfSuspended
      resumeWindow(pid)
    else
      suspendWindow(pid)
  else
    tryResumeWindow(last)
```

**Comment by Merrit** [comment]
In my testing it is frequently (though not always) not possible to focus a suspended window on both linux & windows, making it impossible to "get active window" on it.

I had been considering the possibility of a settings toggle:
- Single suspend mode (default)
- Multi suspend mode

Multi suspend mode would add an entry to the tray menu whenever an application is suspended, and you can click that entry to resume that process. This would ensure you never have an issue being able to find the window if it is unresponsive.

Perhaps even a secondary (optional?) hotkey that opens the menu so you don't have to find it in the tray with your mouse.

**Comment by BlueManCZ** [comment]
Yes, that's a possibility. But it would be great to implement something like I said as well (Quick multi suspend mode?), for users with a desktop environment, that allows focusing unresponsive windows. It would be much easier to use. I use Gnome.

**Comment by Merrit** [comment]
I suppose if you can confirm it is reliable on for example Gnome, there could be a simple check ala:

```
if desktopEnv == GNOME {
  quickMultiMode()
} else {
  singleSuspendMode()
}
```

I have only tested on KDE & Windows 10 and had occasional issues with both.

**Comment by BlueManCZ** [comment]
Yes, this should do it. But it still needs testing.

**Comment by Merrit** [comment]
Suspending multiple applications is available (for Linux) as of https://github.com/Merrit/nyrna/releases/tag/v2.0-alpha.1 & f225f312c1db84dbdcfc81b10a13043f49df7408

Same will be available for Windows as of the forthcoming v2.0-beta.1

**Comment by BlueManCZ** [comment]
Nice, I'm looking forward to it. Could you please update the readme and include steps to compile Nyrna from the source and specify compilation dependencies? I would like to test it.

**Comment by Merrit** [comment]
Sure thing!

https://github.com/Merrit/nyrna/blob/master/COMPILING.MD

Let me know if there is anything else you need or if you get stuck.

**Comment by BlueManCZ** [comment]
I managed to compile this for the first time. Works like a charm. I will probably not be able to create .ebuild for building from source for Gentoo, but a binary version will work. Looking forward to a complete release!

And a funny coincidence, today I hit your [profile](https://blender.stackexchange.com/users/81906/merritt) on StackExchange. :smile:

**Comment by Merrit** [comment]
Heh, small world eh? :grinning: 

If you have any questions about building Flutter, let me know. I also [updated the compiling instructions](https://github.com/Merrit/nyrna/blob/dev/COMPILING.MD) that you can see on the `dev` branch (expecting to merge into master soon, preparing for beta release with Windows support imminently).

**Comment by Merrit** [comment]
Suspending multiple applications is implemented for Linux & Windows as of #28 :smiley:
